### PR TITLE
add sre for app and platform expection

### PR DESCRIPTION
**Pull request by w8896699**

### CODE REVIEW

**Comment by Pilargit12** [comment]
@ShellyXueHan please hold off until my review is done as I will update this PR - thank you so much!

File: src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+- **SLI**: It'll complete 99.5% of data retrieval requests in less than 500 milliseconds
+- **Reason**: Users expect a quick response when querying the registry. Swift data retrieval ensures user satisfaction and efficient downstream operations
+
+**3. Dashboard load time for normal users**
+
+- **SLI**: It'll successfully process 98% of approved product provisioning requests without errors
+- **Reason**: Users rely on the system to seamlessly provision products. A high success rate ensures the reliability of the registry's provisioning functionality
+
+
+<!-- Read more about SLI monitoring setup and available tools  -->
```
Comment (Pilargit12) [inline src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md:106]:
```suggestion
Read more about [SLI monitoring setup and available tools]((../app-monitoring/guidelines-for-sli-and-monitoring.md)
```

Comment (Pilargit12) [reply src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md:106]: Sorry @w8896699 - forgot to include this link, we chatted about yesterday! Adding it here as a review

File: mkdocs.yml [hunk 2]
```diff
@@ -12,7 +12,9 @@ nav:
   - Resource monitoring dashboards: docs/app-monitoring/resource-monitoring-dashboards.md
   - Check application health after an outage: docs/app-monitoring/check-application-health-after-outage.md
   - User Defined Monitoring in OpenShift: docs/app-monitoring/user-defined-monitoring.md
-  - Best practices for application logging in OpenShift: docs/app-monitoring/best-pratices-for-application-logging-in-openshift.md
+  - Best practices for application logging in OpenShift: docs/app-monitoring/best-practices-for-application-logging-in-openshift.md
+  - SRE guidelines for platform shared services: docs/app-monitoring/sre-guidelines-for-platform-shared-services.md
+  - Guidelines for SLI and monitoring: docs/guidelines-for-sli-and-monitoring.md
```
Comment (Pilargit12) [inline mkdocs.yml:17]:
```suggestion
  - Guidelines for SLI and monitoring: docs/app-monitoring/guidelines-for-sli-and-monitoring.md
```

Comment (Pilargit12) [reply mkdocs.yml:17]: Link wasn't working properly - needed to add the category first to be indexed in the left navigation. Apologies!

File: src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+In case we need to adjust these objectives or schedule maintenance windows, we must communicate the reasons for doing so in the [#internal-devops-registry](https://chat.developer.gov.bc.ca/group/internal-devops-registry) Rocket.Chat channel.
+
+**The frequency of backups** such as every 30 minutes, directly links to a system's Recovery Point Objective (RPO), which serves as a critical metric in disaster recovery and business continuity planning. RPO establishes the maximum allowable data loss, measured in time. 
+
+It addresses the question: "How much data can we lose before it starts affecting our business operations?" The reason for configuring it at a 30-minute interval is due to the unique nature of the Platform Product Registry app. 
+
+The Platform Product Registry typically "backs up" most of its crucial data in GitHub repositories, and the provisioner usually processes each request in less than 30 minutes.
+
+**Why is this important?**
```
Comment (ShellyXueHan) [inline src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md:150]: it feels a bit unrelated here. You setup a 30min RPO because of the SLO `The database should have a backup created every 30 minutes`?

File: src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+For more infomation, it is highly recommended to read this [documentation](https://www.atlassian.com/incident-management/kpis/error-budget) about error budget and incident management from Atlassian.
+
+## Conclusion
+In this document, we've covered several essential aspects of Site Reliability Engineering (SRE). From delving into monitoring in detail to implementing tools like Uptime.com. our focus is on ensuring the continuous and smooth operation of our systems.
+
+**Moving forward**:
+1. **Explore the Tools**: If you haven't already, get hands-on experience with the tools we discussed. This will give you a better understanding of how they fit into the bigger picture
+
+2. **Share Feedback**: If you try something out and it works (or doesn't),  share your experience on Rocketchat. We can work together to figure it out. The more we exchange knowledge, the better we become.
+Stay Informed: SRE is a rapidly evolving field. Keep an eye out for updates, new tools, and innovative techniques
```
Comment (ShellyXueHan) [inline src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md:190]:
```suggestion
3. **Stay Informed**: SRE is a rapidly evolving field. Keep an eye out for updates, new tools, and innovative techniques
```

File: src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+For more infomation, it is highly recommended to read this [documentation](https://www.atlassian.com/incident-management/kpis/error-budget) about error budget and incident management from Atlassian.
+
+## Conclusion
+In this document, we've covered several essential aspects of Site Reliability Engineering (SRE). From delving into monitoring in detail to implementing tools like Uptime.com. our focus is on ensuring the continuous and smooth operation of our systems.
+
+**Moving forward**:
+1. **Explore the Tools**: If you haven't already, get hands-on experience with the tools we discussed. This will give you a better understanding of how they fit into the bigger picture
+
+2. **Share Feedback**: If you try something out and it works (or doesn't),  share your experience on Rocketchat. We can work together to figure it out. The more we exchange knowledge, the better we become.
```
Comment (ShellyXueHan) [inline src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md:189]: is there a recommended channel for this?

File: src/docs/platform-architecture-reference/sre-reference-for-app-on-differet-cluster.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+- **Mitigating downtime impact**: Applications in the Gold cluster with failover settings are designed to maintain high availability, even if Vault or Artifactory is temporarily unavailable. By correctly configuring application deployments (e.g., caching secrets or pre-pulling images), teams can ensure that their applications remain operational without needing real-time access to these services during outages
+
+## SLA calculation guidance
+
+### Factors influencing SLAs
+- **Cluster type**: The base SLA varies significantly depending on whether the application is hosted on Silver, Gold, or Emerald
+- **Node configuration**: Single-node vs. multi-node configurations impact the SLA due to differences in failover capabilities
+- **Dependencies**: External dependencies like Vault and Artifactory influence application reliability and need to be considered in SLA calculations
+
+### Factor than need to be considered in your SLA
```
Comment (ShellyXueHan) [inline src/docs/platform-architecture-reference/sre-reference-for-app-on-differet-cluster.md:30]:
```suggestion
### Factors that need to be considered in your SLA
```

File: src/docs/platform-architecture-reference/sre-reference-for-app-on-differet-cluster.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+To calculate the SLA for your application:
```
Comment (ShellyXueHan) [inline src/docs/platform-architecture-reference/sre-reference-for-app-on-differet-cluster.md:31]: can you provide an example on the calculation?

File: src/docs/platform-architecture-reference/sre-reference-for-app-on-differet-cluster.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+## Monitoring and alerting
+- **Monitoring**: Implement comprehensive monitoring that covers all aspects of application performance and health within the cluster environment
+- **Alerting**: Set up alerts based on predefined thresholds to ensure timely responses to potential issues affecting SLA commitments
+
+## Maintaining SLAs
+- **Regular reviews**: SLAs should be reviewed and adjusted annually or after significant changes to application architecture or cluster configuration
+
+- **Collaborators engagement**: Keep collaborators and partners informed about SLA status and any potential impacts from planned or unplanned changes
+
+You can read more about how to setup SRE on an application here
```
Comment (ShellyXueHan) [inline src/docs/platform-architecture-reference/sre-reference-for-app-on-differet-cluster.md:50]: missing link

File: src/docs/platform-architecture-reference/sre-reference-for-app-on-differet-cluster.md (hunk 7 above)
Comment (w8896699) [reply src/docs/platform-architecture-reference/sre-reference-for-app-on-differet-cluster.md:31]: Updated

File: src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md [hunk 9]
```diff
@@ -186,11 +192,9 @@ In this document, we've covered several essential aspects of Site Reliability En
 **Moving forward**:
 1. **Explore the Tools**: If you haven't already, get hands-on experience with the tools we discussed. This will give you a better understanding of how they fit into the bigger picture
 
-2. **Share Feedback**: If you try something out and it works (or doesn't),  share your experience on Rocketchat. We can work together to figure it out. The more we exchange knowledge, the better we become.
+2. **Share Feedback**: If you try something out and it works (or doesn't),  share your experience on Rocketchat[ #sre-discussion](https://chat.developer.gov.bc.ca/channel/sre-discussion). We can work together to figure it out. The more we exchange knowledge, the better we become.
```
Comment (Pilargit12) [inline src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md:195]:
```suggestion
2. **Share feedback**: If you try something out and it works (or doesn't),  share your experience on Rocketchat[ #sre-discussion](https://chat.developer.gov.bc.ca/channel/sre-discussion). We can work together to figure it out. The more we exchange knowledge, the better we become.
```

File: src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md [hunk 10]
```diff
@@ -186,11 +192,9 @@ In this document, we've covered several essential aspects of Site Reliability En
...
 3. **Stay Informed**: SRE is a rapidly evolving field. Keep an eye out for updates, new tools, and innovative techniques
```
Comment (Pilargit12) [inline src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md:196]:
```suggestion
3. **Stay informed**: SRE is a rapidly evolving field. Keep an eye out for updates, new tools, and innovative techniques
```

File: src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md [hunk 11]
```diff
@@ -28,16 +28,17 @@ In this document, we'll explore the fundamental concept behind SRE and show you
 ## On this page
 
 - [SRE guidelines for platform shared services](#sre-guidelines-for-platform-shared-services)
```
Comment (Pilargit12) [inline src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md:30]:
```suggestion
- [SRE guidelines for platform shared services](#sre-guidelines-for-platform-shared-services)
- [B.C. Platform Services Product Registry](#bc-platform-services-product-registry)
- [Setting up SRE](#setting-up-sre)
- [Importance of defining a period for SRE measurement](#importance-of-defining-a-period-for-sre-measurement)
- [Service Level Indicators (SLIs)](#service-level-indicators-slis)
- [Service Level Agreement (SLA)](#service-level-agreement-sla)
- [Service Level Objective (SLO)](#service-level-objective-slo)
- [Connections and differences](#connections-and-differences)
- [How to calculate error budget](#how-to-calculate-error-budget)
- [Conclusion](#conclusion)
- [Related pages](#related-pages)
```

Comment (Pilargit12) [reply src/docs/app-monitoring/sre-guidelines-for-platform-shared-services.md:30]: All same level H2 headings appear as menu items. 

Always remove "On this page" heading - not used
